### PR TITLE
fix transfer sampled and parametrized rf modules to cuda

### PR DESCRIPTION
**Pull request by dimitriskor**
ParameterizedSpatialReceptiveField2d and SampledSpatialReceptiveField2d where not transferring submodule's weights  to cuda.

Fixed by registering the weights as buffers.

### CODE REVIEW

**Review by Jegp — APPROVED** [review]
That's a great addition @dimitriskor, thank you :tada:

File: norse/torch/module/receptive_field.py [hunk 1]
```diff
@@ -112,7 +114,9 @@ def _update_weights(self):
                         in_weights = empty_weights.clone()
                         in_weights[i] = fields
                         weights.append(in_weights)
-                    self.weights = torch.concat(weights, 1).permute(1, 0, 2, 3)
+                    self.register_buffer(
```
Comment (Jegp) [inline norse/torch/module/receptive_field.py:117]: @dimitriskor, I just realized that buffers are not considered a model parameter. Should this have been `register_parameter`? https://pytorch.org/docs/stable/generated/torch.nn.Module.html#torch.nn.Module

Comment (dimitriskor) [reply norse/torch/module/receptive_field.py:117]: The decision for declaring the weights as buffers has been deliberate, as they are only part of the computational graph and not leafs.
The grad is set to True and the gradient computation flows backwards to the rf parameters which are the leaf nodes of the computational graph.
